src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -620,3 +636,250 @@ TEST_F(ControllerScriptEngineLegacyTest, connectionExecutesWithCorrectThisObject
     // The counter should have been incremented exactly once.
     EXPECT_DOUBLE_EQ(1.0, pass->get());
 }
+
+TEST_F(ControllerScriptEngineLegacyTimerTest, beginTimer_repeatedTimer) {
+    EXPECT_TRUE(evaluateAndAssert(
+            "engine.setValue('[Test]', 'co', 0.0);"));
+    EXPECT_DOUBLE_EQ(0.0, co->get());
+
+    EXPECT_TRUE(
+            evaluateAndAssert(R"(engine.beginTimer(50, function() {
+                                    let x = engine.getValue('[Test]', 'co');
+                                    x++; 
+                                    engine.setValue('[Test]', 'co', x);
+                                 }, false);)"));
+    processEvents();
+    EXPECT_DOUBLE_EQ(0.0, co->get());
+
+    cEngine->thread()->msleep(70);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(1.0, co->get());
+
+    cEngine->thread()->msleep(140);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(2.0, co->get());
+}
+
+TEST_F(ControllerScriptEngineLegacyTimerTest, beginTimer_singleShotTimer) {
+    EXPECT_TRUE(evaluateAndAssert(
+            "engine.setValue('[Test]', 'co', 0.0);"));
+    EXPECT_DOUBLE_EQ(0.0, co->get());
+
+    // Single shot timer with minimum allowed interval of 20ms
+    EXPECT_TRUE(evaluateAndAssert(
+            R"(engine.beginTimer(20, function() {
+                   engine.setValue('[Test]', 'co', 1.0);
+               }, true);)"));
+    processEvents();
+    EXPECT_DOUBLE_EQ(0.0, co->get());
+
+    cEngine->thread()->msleep(35);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(1.0, co->get());
+}
+
+TEST_F(ControllerScriptEngineLegacyTimerTest, beginTimer_singleShotTimerBindFunction) {
+    // Single shot timer with minimum allowed interval of 20ms
+    EXPECT_TRUE(evaluateAndAssert(
+            R"(var globVar = 7;
+            timerId = engine.beginTimer(20, function () {
+                engine.setValue('[Test]', 'co', this.globVar);
+                this.globVar++;
+                engine.setValue('[Test]', 'coTimerId', timerId + 10);
+            }.bind(this), true);            
+            engine.setValue('[Test]', 'coTimerId', timerId);)"));
+    processEvents();
+    EXPECT_DOUBLE_EQ(0.0, co->get());
+    double timerId = coTimerId->get();
+    EXPECT_TRUE(timerId > 0);
+
+    cEngine->thread()->msleep(35);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(timerId + 10, coTimerId->get());
+    EXPECT_DOUBLE_EQ(7.0, co->get());
+    EXPECT_TRUE(evaluateAndAssert("engine.setValue('[Test]', 'co', this.globVar);"));
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(8.0, co->get());
+}
+
+TEST_F(ControllerScriptEngineLegacyTimerTest, beginTimer_singleShotTimerArrowFunction) {
+    // Single shot timer with minimum allowed interval of 20ms
+    EXPECT_TRUE(evaluateAndAssert(
+            R"(var globVar = 7;
+            timerId = engine.beginTimer(20, ()=> {
```

Comment:
Small formatting nitpick.
```suggestion
            timerId = engine.beginTimer(20, () => {
```